lib/Producer.d.ts

```diff
@@ -14,6 +14,7 @@ export declare type ProducerCodecOptions = {
     opusStereo?: boolean;
     opusFec?: boolean;
     opusDtx?: boolean;
+    opusNack?: boolean;
```

Comment:
Rest of opusXXXX parameters are really opus parameters but the nack one is not a codec parameter but a transport one.   It shouldn't be a codec option  IMO.

Reply:
The thing here is that, in order to make libwebrtc offer NACK in OPUS, we need to mangle the SDP offer.

Do you mean that it should be codec agnostic, something such as `forceNack` instead?

Reply:
Also if there is a different place for it instead of ProducerCodecOption structure it would be better given that this is not really a codec parameter.   But I'm not sure if this is doable in the current design.

Reply:
RTCP feedbacks are somehow "codec options" so I think the option is fine here.

However I need to properly review this PR once I get time since I'm not fully convinced about how it's done. So please be patient.

Reply:
Why do you folks think that bad SDP munging is required? On the [sample page](https://webrtc.github.io/samples/src/content/peerconnection/audio/) try this (with loss)
```
await pc1.setLocalDescription();
await pc2.setRemoteDescription({type: 'offer', sdp: pc1.localDescription.sdp +  'a=rtcp-fb:111 nack\r\n'});
await pc2.setLocalDescription();
await pc1.setRemoteDescription({type: 'answer', sdp: pc2.localDescription.sdp +  'a=rtcp-fb:111 nack\r\n'});
```
and then observe the `retransmittedPacketsSent` on the outbound-rtp stats increase.
No munging, just some SDP manipulation which is totally allowed.

Reply:
What? We do it much better by modifying the specific m= section we want to affect and not just the last one or the whole SDP.

Reply:
The point is that you don't even need clientside support. The only browser not supporting this is Firefox and what happens if the server announces nack is simply that Firefox won't do resends. Easier to infer from the UA than putting it into caps

Reply:
Definitely we don't want that mediasoup sends NACK to a device that doesn't support NACK for the given codec.